File: ontrack-extension-svn/src/main/java/net/nemerosa/ontrack/extension/svn/db/SVNRepository.java
package net.nemerosa.ontrack.extension.svn.db;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import net.nemerosa.ontrack.extension.issues.model.ConfiguredIssueService;
import net.nemerosa.ontrack.extension.svn.model.SVNConfiguration;
import net.nemerosa.ontrack.extension.svn.support.SVNUtils;
import org.tmatesoft.svn.core.SVNURL;

@Data
@AllArgsConstructor(access = AccessLevel.PROTECTED)
public class SVNRepository {

    private final int id;
    private final SVNConfiguration configuration;
    private final ConfiguredIssueService configuredIssueService;

    public static SVNRepository of(int id, SVNConfiguration configuration, ConfiguredIssueService configuredIssueService) {
        return new SVNRepository(id, configuration, configuredIssueService);
    }

    public String getBranchPattern() {
        return ".*/branches/[^/]+";
    }

    public String getTagPattern() {
        return ".*/tags/[^/]+";
    }

    public String getUrl(String path) {
        return configuration.getUrl(path);
    }

    public String getRevisionBrowsingURL(long revision) {
        return configuration.getRevisionBrowsingURL(revision);
    }

    public String getPathBrowsingURL(String path) {
        return configuration.getPathBrowsingURL(path);
    }

    public String getFileChangeBrowsingURL(String path, long revision) {
        return configuration.getFileChangeBrowsingURL(path, revision);
    }

    public SVNURL getRootUrl() {
        return SVNUtils.toURL(configuration.getUrl());
    }
}

File: ontrack-extension-svn/src/main/java/net/nemerosa/ontrack/extension/svn/db/SVNRepository.kt
package net.nemerosa.ontrack.extension.svn.db

import net.nemerosa.ontrack.extension.issues.model.ConfiguredIssueService
import net.nemerosa.ontrack.extension.svn.model.SVNConfiguration
import net.nemerosa.ontrack.extension.svn.support.SVNUtils
import org.tmatesoft.svn.core.SVNURL

class SVNRepository(
        val id: Int,
        val configuration: SVNConfiguration,
        val configuredIssueService: ConfiguredIssueService? = null
) {

    val branchPattern: String
        get() = ".*/branches/[^/]+"

    val tagPattern: String
        get() = ".*/tags/[^/]+"

    val rootUrl: SVNURL
        get() = SVNUtils.toURL(configuration.url)

    fun getUrl(path: String): String {
        return configuration.getUrl(path)
    }

    fun getRevisionBrowsingURL(revision: Long): String {
        return configuration.getRevisionBrowsingURL(revision)
    }

    fun getPathBrowsingURL(path: String): String {
        return configuration.getPathBrowsingURL(path)
    }

    fun getFileChangeBrowsingURL(path: String, revision: Long): String {
        return configuration.getFileChangeBrowsingURL(path, revision)
    }

    companion object {

        @JvmStatic
        fun of(id: Int, configuration: SVNConfiguration, configuredIssueService: ConfiguredIssueService): SVNRepository {
            return SVNRepository(id, configuration, configuredIssueService)
        }
    }
}
